scene/resources/visual_shader_nodes.cpp

```diff
@@ -3742,10 +3748,24 @@ VisualShaderNodeDerivativeFunc::Function VisualShaderNodeDerivativeFunc::get_fun
 	return func;
 }
 
+void VisualShaderNodeDerivativeFunc::set_precision(Precision p_precision) {
+	ERR_FAIL_INDEX(int(p_precision), int(PRECISION_MAX));
+	if (precision == p_precision) {
```

Comment:
We should also have a check here to ensure the setting isn't changed when using the gl_compatibility renderer as it doesn't support the coarse/fine variants.

Reply:
Do you have an idea of how to make this check? Otherwise, since the visual shader is compiled into a shader and I mark these functions as high-level - the error will be occurred if the user tried to use it on compatibility profile anyway.

Reply:
Elsewhere in the engine we use
```
if (OS::get_singleton()->get_current_rendering_method() == "gl_compatibility") {
```
I guess we will run into problems if users create a visual shader in one of the RD backends and then export to GL_Compatibility. But a check here is better than nothing.